<launch>
<!-- Launch file for Livox AVIA LiDAR -->

	<arg name="rviz" default="true" />

	<rosparam command="load" file="$(find ige_lio)/config/ige_lio.yaml" />

	<param name="feature_extract_enable" type="bool" value="1"/>

	<param name="edge_plane_flag" type="int" value="3"/>
	<param name="point_filter_num" type="int" value="2"/>               
	<param name="max_iteration" type="int" value="4" />

	<param name="filter_size_map" type="double" value="0.3" />

	<param name="filter_size_surf" type="double" value="0.5" />
	<param name="filter_size_edge" type="double" value="0.1" />
	<param name="filter_size_inten_edge" type="double" value="0.1" />

	<param name="cube_side_length" type="double" value="1000" />
    <node pkg="ige_lio" type="igelio_mapping" name="laserMapping" output="screen" /> 

	<group if="$(arg rviz)">
	<node launch-prefix="nice" pkg="rviz" type="rviz" name="rviz" args="-d $(find ige_lio)/rviz_cfg/loam_livox.rviz" />
	</group>

</launch>
